contrib/pkg/provision/command.go

```diff
@@ -0,0 +1,206 @@
+package provision
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	capiclient "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+
+	clustopclient "github.com/openshift/cluster-operator/pkg/client/clientset_generated/clientset"
+)
+
+type ProvisionClusterOptions struct {
+	Namespace           string
+	Name                string
+	ImageFormat         string
+	RPMRepositoryURL    string
+	RPMRepositorySource string
+	SSHSecretName       string
+	AccountSecretName   string
+	KeyPairName         string
+	AnsibleImage        string
+	Region              string
+	KubeconfigFile      string
+	Version             string
+	VPCName             string
+	LogLevel            string
+
+	ImageID string
+	Logger  log.FieldLogger
+}
+
+func NewProvisionClusterCommand() *cobra.Command {
+	userName := "user"
+	u, err := user.Current()
+	if err == nil {
+		userName = u.Username
+	}
+
+	opt := &ProvisionClusterOptions{}
+	cmd := &cobra.Command{
+		Use:   "provision NAME",
```

Comment:
This is generically named provision but when we make it to multi-cloud, would it be good to just rename the top level command to provision-aws and we can add more as supported?

Reply:
Might make sense to apply to the package name as well if we go ahead with that.

Reply:
Makes sense, will update